pkg/controller/dnsendpoint/ingress.go

```diff
@@ -19,29 +19,22 @@ package dnsendpoint
 import (
 	"github.com/pkg/errors"
 
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	pkgruntime "k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/watch"
 	restclient "k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	feddnsv1a1 "github.com/kubernetes-sigs/federation-v2/pkg/apis/multiclusterdns/v1alpha1"
-	fedclientset "github.com/kubernetes-sigs/federation-v2/pkg/client/clientset/versioned"
 	"github.com/kubernetes-sigs/federation-v2/pkg/controller/util"
 )
 
 func StartIngressDNSEndpointController(config *util.ControllerConfig, stopChan <-chan struct{}) error {
 	restclient.AddUserAgent(config.KubeConfig, "Ingress DNSEndpoint")
-	client := fedclientset.NewForConfigOrDie(config.KubeConfig)
-
-	listFunc := func(options metav1.ListOptions) (pkgruntime.Object, error) {
-		return client.MulticlusterdnsV1alpha1().IngressDNSRecords(config.TargetNamespace).List(options)
-	}
-	watchFunc := func(options metav1.ListOptions) (watch.Interface, error) {
-		return client.MulticlusterdnsV1alpha1().IngressDNSRecords(config.TargetNamespace).Watch(options)
+	client, err := client.New(config.KubeConfig, client.Options{})
```

Comment:
Consider initializing the client closer to where it is used - in `newDNSEndpointController` - since there is no longer any other use in this method.